src/plugins/minimal_scene/MinimalScene.cc

```diff
@@ -431,31 +436,65 @@ void IgnRenderer::BroadcastRightClick()
 
   events::RightClickToScene rightClickToSceneEvent(pos);
   App()->sendEvent(App()->findChild<MainWindow *>(), &rightClickToSceneEvent);
+
   events::RightClickOnScene rightClickOnSceneEvent(this->dataPtr->mouseEvent);
   App()->sendEvent(App()->findChild<MainWindow *>(), &rightClickOnSceneEvent);
+
+  this->dataPtr->mouseDirty = false;
+}
+
+/////////////////////////////////////////////////
+void IgnRenderer::BroadcastMousePress()
+{
+  if (!this->dataPtr->mouseDirty)
+    return;
+
+  if (this->dataPtr->mouseEvent.Type() != common::MouseEvent::PRESS)
+    return;
+
+  events::MousePressOnScene event(this->dataPtr->mouseEvent);
+  App()->sendEvent(App()->findChild<MainWindow *>(), &event);
+
+  this->dataPtr->mouseDirty = false;
 }
 
+/////////////////////////////////////////////////
+void IgnRenderer::BroadcastScroll()
+{
+  if (!this->dataPtr->mouseDirty)
+    return;
+
+  if (this->dataPtr->mouseEvent.Type() != common::MouseEvent::SCROLL)
+    return;
+
+  events::ScrollOnScene scrollOnSceneEvent(this->dataPtr->mouseEvent);
+  App()->sendEvent(App()->findChild<MainWindow *>(), &scrollOnSceneEvent);
+
+  this->dataPtr->mouseDirty = false;
+}
 
 /////////////////////////////////////////////////
 void IgnRenderer::BroadcastKeyRelease()
 {
-  if (this->dataPtr->keyEvent.Type() == common::KeyEvent::RELEASE)
-  {
-    events::KeyReleaseOnScene keyRelease(this->dataPtr->keyEvent);
-    App()->sendEvent(App()->findChild<MainWindow *>(), &keyRelease);
-    this->dataPtr->keyEvent.SetType(common::KeyEvent::NO_EVENT);
-  }
+  if (this->dataPtr->keyEvent.Type() != common::KeyEvent::RELEASE)
+    return;
+
+  events::KeyReleaseOnScene keyRelease(this->dataPtr->keyEvent);
+  App()->sendEvent(App()->findChild<MainWindow *>(), &keyRelease);
+
+  this->dataPtr->keyEvent.SetType(common::KeyEvent::NO_EVENT);
 }
 
 /////////////////////////////////////////////////
 void IgnRenderer::BroadcastKeyPress()
 {
   if (this->dataPtr->keyEvent.Type() == common::KeyEvent::PRESS)
```

Comment:
Should this be `!=`?
```suggestion
  if (this->dataPtr->keyEvent.Type() != common::KeyEvent::PRESS)
```

Reply:
since the next line is a `return`

Reply:
oh yeah! nice catch